core/src/replay_stage.rs

```diff
@@ -3618,6 +3623,8 @@ impl ReplayStage {
                 //
                 // For non leader banks (2) is always true, so notify here
                 if *is_alpenglow_migration_complete && bank.block_id().is_some() {
+                    // XXX: this is the leader producing the block?
```

Comment:
The block associated with `bank` may or may not have been produced by this node. We can check by comparing `bank.collector_id` against our `pubkey`